lz4.go

```diff
@@ -16,7 +16,12 @@ type Lz4 struct {
 // Compress reads in, compresses it, and writes it to out.
 func (lz *Lz4) Compress(in io.Reader, out io.Writer) error {
 	w := lz4.NewWriter(out)
-	w.Header.CompressionLevel = lz.CompressionLevel
+	options := []lz4.Option{
+		lz4.CompressionLevelOption(lz4.CompressionLevel(1 << (8 + lz.CompressionLevel))),
```

Comment:
Same comment here as below. Default compression should not need to be set, but if we want to set it to be explicit (which I think we do), it should be `lz4.Fast`.

Reply:
Ah, I see why this was done. This way does preserve backwards compatibility, but I think it deserves a comment, something like:

```
// TODO archiver v4: use proper lz4.Fast
// bitshifting for backwards compatibility with lz4/v3
```

Reply:
Done,